internal/persistence/sql/traverser.go

```diff
@@ -127,6 +127,8 @@ LIMIT ?
 }
 
 func (t *Traverser) TraverseSubjectSetRewrite(ctx context.Context, start *relationtuple.RelationTuple, computedSubjectSets []string) (res []*relationtuple.TraversalResult, err error) {
+	ctx, span := t.d.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.TraverseSubjectSetRewrite")
+	defer otelx.End(span, &err)
```

Comment:
I get why you want to put the tracing stuff at the top, but the reason why I put it in front of the query was because sometimes the query does not even execute, and I want to have some way to count the number of DB queries that resulted from one `check` request.

If you want to put the tracing up top I think we should extract the query out into its own function that then has the tracing up top. WDYT?

Reply:
You could also just add an attribute to the span once you run into that case. Would be a bit better IMO.

Reply:
The DB query (if it happens) would get it's own child spawn though, wouldn't it?